ex5/MyStockApplication/app/src/main/java/com/example/mystockapplication/MyFirebaseMessagingService.java

```diff
@@ -0,0 +1,107 @@
+package com.example.mystockapplication;
+
+import android.app.Activity;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.media.RingtoneManager;
+import android.net.Uri;
+import android.os.Build;
+import android.os.IBinder;
+import android.util.Log;
+import android.widget.TextView;
+
+import androidx.core.app.NotificationCompat;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.Volley;
+import com.google.firebase.messaging.FirebaseMessagingService;
+import com.google.firebase.messaging.RemoteMessage;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class MyFirebaseMessagingService extends FirebaseMessagingService {
+    private static final String TAG = "MyFirebaseService";
+    private static final String SERVER_ADDRESS = "http://10.26.181.52:3000/";
```

Comment:
Think about moving this one to configuration so you won't have to re-compile when changing the ip